### PR TITLE
Unit tests

### LINKED ISSUE #131: Testing framework
As a contributor, I'd like a testing framework so that I can be more confident that my changes work as expected (and don't introduce bugs).

[`pytest`](https://docs.pytest.org/) seems like a logical choice.

### LINKED ISSUE #132: Contributing instructions
As a contributor, I'd like instructions so that I know how to do development and test my work.

### PR DESCRIPTION
**Pull request by dbeatty10**
Resolves #131, resolves #132, related to #67, supports #130

## Overview
- Simple test cases for validating package versions specified in git tags
- Brief instructions for installing dependencies and running test cases
- We will need to change the test cases slightly if #130 is merged

## Implementation details
Used `pytest` as the testing framework.

There was a naming conflict between the local `hubcap/cmd.py` module and the [`git/cmd.py`](https://github.com/gitpython-developers/GitPython/blob/f23994e3bdd20164a1365296153044d5c7718fef/git/cmd.py) module within the `gitpython` package. So I renamed the former.

Used [these](https://regex101.com/r/Ly7O1x/3/) test cases (linked from the [SemVer docs](https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string)). Added a couple test cases we've seen for dbt adapters and packages.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+### Config
+
+```shell
+cp config.example.json config.json
+
+# Add your GitHub username and token
+$EDITOR config.json
+
+# Export the JSON credentials into an environment variable
+export CONFIG=$(<config.json)
```
Comment (joellabes) [inline CONTRIBUTING.md:23]: I'm unfamiliar with this syntax, so just checking that the lonely `<` is meant to be there?

**Review by joellabes — APPROVED** [review]
in the most literal way possible, this looks good to me!

File: tests/test_version_regex.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        "-1.0.3-gamma+b7718",
+        "+justmeta",
+        "9.8.7+meta+meta",
+        "9.8.7-whatever+meta+meta",
+        "99999999999999999999999.999999999999999999.99999999999999999----RC-SNAPSHOT.12.09.1--------------------------------..12",
+        # Begin custom test cases
+        "1.0.3.post1",
+        "0.8.0a1",
+        "0.8.0b2",
+        "0.8.0rc3",
```
Comment (jtcohen6) [inline tests/test_version_regex.py:100]: 👍

#130 will move these up to the "supported" category

**Review by jtcohen6 — APPROVED** [review]
looks fantastic!

**Pull request merged**